client/dashboard/me/billing-purchases/cancel-purchase/cancel-purchase-form/step-components/upsell-step.tsx

```diff
@@ -208,9 +208,10 @@ export default function UpsellStep( {
 					acceptButtonUrl={ wpcomLink(
 						`/checkout/${
 							purchase.site_slug
-						}/business?coupon=${ couponCode }&cancel_to=${ window.location.href
-							.replace( window.location.origin, '' )
-							.replace( '/cancel', '' ) }`
+						}/business?coupon=${ couponCode }&cancel_to=${ redirectToDashboardLink().replace(
```

Comment:
This can be tested by going to the subscription page of a Personal/Premium plan, and select one of the "missing feature" options such as cannot install plugin/theme. The upsell page will contain this link.